di-ipv-core-stub/src/main/java/uk/gov/di/ipv/stub/core/config/uatuser/IdentityMapper.java

```diff
@@ -123,23 +156,72 @@ public Identity mapFormToIdentity(Identity identityOnRecord, QueryParamsMap user
                         null,
                         userData.value("townCity"),
                         userData.value("postCode"),
-                        true);
-
-        int year = Integer.parseInt(userData.value("dateOfBirth-year"));
-        int month = Integer.parseInt(userData.value("dateOfBirth-month"));
-        int dayOfMonth = Integer.parseInt(userData.value("dateOfBirth-day"));
+                        vf,
+                        vu);
+        addresses.add(ukAddress);
+
+        LocalDate secondaryVf =
+                getLocalDate(
+                        userData,
+                        "SecondaryUKAddress.validFrom-year",
+                        "SecondaryUKAddress.validFrom-month",
+                        "SecondaryUKAddress.validFrom-day");
+
+        LocalDate secondaryVu =
+                getLocalDate(
+                        userData,
+                        "SecondaryUKAddress.validUntil-year",
+                        "SecondaryUKAddress.validUntil-month",
+                        "SecondaryUKAddress.validUntil-day");
+
+        UKAddress secondaryUkAddress =
+                new UKAddress(
+                        userData.value("SecondaryUKAddress.buildingNumber"),
+                        userData.value("SecondaryUKAddress.buildingName"),
+                        userData.value("SecondaryUKAddress.street"),
+                        null,
+                        userData.value("SecondaryUKAddress.townCity"),
+                        userData.value("SecondaryUKAddress.postCode"),
+                        secondaryVf,
+                        secondaryVu);
+        if (!Stream.of(
```

Comment:
move this `if` conditional statement further up and check the form data for the presence of some of the mandatory address fields e.g. street / postcode
Then only parse the dates and instantiate UKAddress if the form data indicates that a secondary address has been entered

Reply:
The reason to check them all is to allow an address to be created with any combination of fields so we can check error scenarios more easily